torchgeo/datasets/digital_typhoon.py

```diff
@@ -0,0 +1,435 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Digital Typhoon dataset."""
+
+import glob
+import os
+import tarfile
+from collections.abc import Sequence
+from typing import Any, Callable, Optional, TypedDict
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import DatasetNotFoundError, download_url, percentile_normalization
+
+
+class _SampleSequenceDict(TypedDict):
+    """Sample sequence dictionary."""
+
+    id: str
+    seq_id: list[int]
+
+
+class DigitalTyphoonAnalysis(NonGeoDataset):
```

Comment:
Let's call it `DigitalTyphoon` to match the name of the dataset on the website.

Reply:
under section 4.1 in their paper https://arxiv.org/pdf/2311.02665.pdf they list the difference inference tasks possible with this dataset, this implementation only covers the "Analysis" task so that is why I made that name choice

Reply:
I guess it depends on how different the other tasks are. If it's possible to write a single dataset for all 3 tasks, I would want to name the dataset DigitalTyphoon and expand it later.